app/models/concerns/hostext/search.rb

```diff
@@ -222,6 +222,36 @@ def search_cast_facts(key, operator, value)
         }
       end
 
+      def search_by_os_major(key, operator, value)
+        condition = sanitize_sql_for_conditions(["CAST(major AS INTEGER) #{operator} ?", value_to_sql(operator, value.to_i)])
+        operatingsystem_ids = Operatingsystem.select(:id).where(condition).pluck('operatingsystems.id').join(',')
+        {:conditions => "hosts.operatingsystem_id IN (#{operatingsystem_ids})"}
+      end
+
+      def search_by_os_minor(key, operator, value)
+        if value.include? '.'
+          minor = value.split('.')[0].to_i
+          patch = value.split('.')[1].to_i
+        else
+          minor = value.to_i
+          patch = 0
+        end
+        operatingsystem_ids = Operatingsystem.find_by_sql(["SELECT id
+                                                            FROM (SELECT *,
+                                                                  CASE WHEN minor like '%.%'
+                                                                  THEN substr(minor, 1, pos-1)
+                                                                  ELSE minor END AS new_minor,
+                                                                  CASE WHEN minor like '%.%'
+                                                                  THEN substr(minor, pos+1)
+                                                                  ELSE '0' END AS patch
+                                                                  FROM (SELECT *,
+                                                                        instr(minor,'.') AS pos
+                                                                        FROM operatingsystems))
+                                                            WHERE CAST(new_minor AS INTEGER) #{operator}= ?
```

Comment:
Why is there `=` operator after the `%{operator}`? :)

Reply:
Because of the separation between the minor and the patch after the dot. For example, when searching for os_minor > 6, 6.1 will not appear without the =. If you have another idea for it I would like to hear, I know it is a bit ugly.

Reply:
Regarding what you wrote before - I know it works well on SQLite, I will make sure it works on the other two supported engines.